qiskit_experiments/curve_analysis/base_curve_analysis.py

```diff
@@ -0,0 +1,550 @@
+
+        Args:
+            fit_data: Fit outcome.
+            quality: Quality of fit outcome.
+
+        Returns:
+            List of analysis result data.
+        """
+        outcomes = []
+
+        # Create entry for all fit parameters
+        if self.options.return_fit_parameters:
+            fit_parameters = AnalysisResultData(
+                name=PARAMS_ENTRY_PREFIX + self.__class__.__name__,
+                value=[p.nominal_value for p in fit_data.popt],
+                chisq=fit_data.reduced_chisq,
+                quality=quality,
+                extra={
+                    "popt_keys": fit_data.popt_keys,
+                    "dof": fit_data.dof,
+                    "covariance_mat": fit_data.pcov,
+                    **metadata,
+                },
+            )
+            outcomes.append(fit_parameters)
+
+        # Create entries for important parameters
+        for param_repr in self.options.result_parameters:
+            if isinstance(param_repr, ParameterRepr):
+                p_name = param_repr.name
+                p_repr = param_repr.repr or param_repr.name
+                unit = param_repr.unit
+            else:
+                p_name = param_repr
+                p_repr = param_repr
+                unit = None
+
+            fit_val = fit_data.fitval(p_name)
+            if unit:
+                par_metadata = metadata.copy()
+                par_metadata["unit"] = unit
+            else:
+                par_metadata = metadata
+
+            outcome = AnalysisResultData(
+                name=p_repr,
+                value=fit_val,
+                chisq=fit_data.reduced_chisq,
+                quality=quality,
+                extra=par_metadata,
+            )
+            outcomes.append(outcome)
+
+        return outcomes
+
+    def _create_curve_data(
+        self,
+        curve_data: CurveData,
+        series: List[SeriesDef],
+        **metadata,
+    ) -> List[AnalysisResultData]:
+        """Create analysis results for raw curve data.
+
+        Args:
+            curve_data: Full curve dataset used for the fitting.
+            series: List of series definition associated with the curve data.
+
+        Returns:
+            List of analysis result data.
+        """
+        samples = []
+
+        if not self.options.return_data_points:
+            return samples
+
+        for sdef in series:
+            s_data = curve_data.get_subset_of(sdef.name)
+            raw_datum = AnalysisResultData(
+                name=DATA_ENTRY_PREFIX + self.__class__.__name__,
+                value={
+                    "xdata": s_data.x,
+                    "ydata": s_data.y,
+                    "sigma": s_data.y_err,
+                },
+                extra={
+                    "name": sdef.name,
+                    **metadata,
+                },
+            )
+            samples.append(raw_datum)
+
+        return samples
+
+    def _preparation(
```

Comment:
This method name is a bit vague. Can it be called something more descriptive? Maybe even `_initalize` is more apt